### PR TITLE
Fix GN dependencies in //brave/utility

### PR DESCRIPTION
**Pull request by mariospr**
We had made services.cc depend on //brave/utility via a chromium_src
override, but that caused a dependency cycle because //brave/utility
also depends on //chrome/utility (e.g. extend some upstream classes),
causing a gn check error.
    
To fix this, we can follow the pattern used in other places and make
upstream's //chrome/utility directly include the files from Brave that
it needs, and replace the oneliner patch to //chrome/utility/BUILD.gn
with one that would simply include those sources, instead of having it
depend on //brave/utility.
    
Last, we also add a missing dep on //brave/common/importer:importer
to avoid another gn check failure due to having some files from that
target included from brave_external_process_importer_bridge.h.
    
Resolves https://github.com/brave/brave-browser/issues/10623

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: build/config/brave_build.gni
Comment (bridiver) [inline build/config/brave_build.gni:21]: this file has actually become a problem in some places because of circular gni includes so I think we should add the import to the patch on the same line as the sources +=, etc...

File: browser/profiles/BUILD.gn
Comment (bridiver) [inline browser/profiles/BUILD.gn:1]: ?

Comment (mariospr) [reply browser/profiles/BUILD.gn:1]: I needed to add this import **not** to get the following error:
```
> gn check /home/mario/work/brave-browser/src/out/Component //brave/*
ERROR at //brave/browser/profiles/BUILD.gn:65:7: Undefined identifier
  if (decentralized_dns_enabled) {
      ^------------------------
See //brave/browser/BUILD.gn:120:5: which caused the file to be included.
    "profiles",
    ^---------
```

And the reason this is needed now is because of removing the `import("//build/config/features.gni")` line from `components/tor/buildflags/buildflags.gni`:

Previous to this patch, `brave/browser/profiles/BUILD.gn` would indirectly satisfy the dependency on `//brave/components/decentralized_dns/buildflags` via [the `//brave/components/tor/buildflags/buildflags.gni` import it already has in place](https://github.com/brave/brave-core/blob/master/browser/profiles/BUILD.gn#L1), which would end up importing that GNI file as it can be seen in the tree below:
```
brave/browser/profiles/BUILD.gn
└─> brave/components/tor/buildflags/buildflags.gni
     └─> build/config/features.gni
          └─> build/config/chrome_build.gni
               └─> brave/build/config/brave_build.gni
                    └─> brave/net/sources.gni
                         └─> brave/components/decentralized_dns/buildflags/buildflags.gni
```
... but now that the `brave/components/tor/buildflags/buildflags.gni ─> build/config/features.gni` link does not exist we need to explicitly add the import to `brave/browser/profiles/BUILD.gn`, or we'll get a new `gn check` failure.

Sorry if this was not clear enough from the PR.

File: build/config/brave_build.gni
Comment (mariospr) [reply build/config/brave_build.gni:21]: SGTM. I actually faced that circular dep issue you mentioned while writing this patch and my first approach was to do exactly this thing you mention here, but then I found out I could avoid that dep cycle by removing those `import("//build/config/features.gni")` lines from the following `buildflags.gni` files:

   components/brave_ads/browser/buildflags/buildflags.gni
   components/brave_rewards/browser/buildflags/buildflags.gni
   components/ipfs/buildflags/buildflags.gni
   components/tor/buildflags/buildflags.gni    

...where there were were not explicitly needed anyway.

I'll do the in-place import then, and I'll drop the removal of those imports in the `buildflags.gni` files then as they are not related, and will propose a cleanup of those imports in all the affected `buildflags.gni` files (and not just on those 4 ones) as a separate PR, if that sounds Ok.

File: browser/profiles/BUILD.gn
Comment (mariospr) [reply browser/profiles/BUILD.gn:1]: Note: after the latest changes I made, described in https://github.com/brave/brave-core/pull/8443#discussion_r608489705, it's not really strictly needed so I've removed this bit as well from this PR and will incorporate it in the new one I'll propose to get rid of those import from all the affected `buildflags.gni` files at once.

File: build/config/brave_build.gni
Comment (mariospr) [reply build/config/brave_build.gni]: JFTR: PR for cleaning up imports from those buildflags.gni files here: https://github.com/brave/brave-core/pull/8461

File: utility/BUILD.gn
Comment (bridiver) [inline utility/BUILD.gn:1]: let's keep this with `group("utility") { public_deps = [ "//chrome/utility" ] }` as discussed in dm

Comment (mariospr) [reply utility/BUILD.gn:1]: Good point. I've changed the PR to do this now.

@bridiver PTAL when you have a moment, thanks!

**Comment by mariospr** [comment]
@bridiver Ping on this one as well for a second review, thanks

**Review by bridiver — APPROVED** [review]

**Comment by mariospr** [comment]
@tmancey Ping on this one (blocked on you as code owner), thanks!

**Review by tmancey — APPROVED** [review]
LGTM

**Comment by mariospr** [comment]
CI errors are unrelated, so merging. Thank you all for the reviews!

**Pull request merged**
